homework_6/task_1.py

```diff
@@ -0,0 +1,252 @@
+— Таблица «Товары» должна содержать информацию о доступных товарах, их
+описаниях и ценах.
+— Таблица «Заказы» должна содержать информацию о заказах, сделанных
+пользователями.
+— Таблица «Пользователи» должна содержать информацию о зарегистрированных
+пользователях магазина.
+• Таблица пользователей должна содержать следующие поля:
+id (PRIMARY KEY),
+имя, фамилия, адрес электронной почты и пароль.
+• Таблица заказов должна содержать следующие поля:
+id (PRIMARY KEY), id
+пользователя (FOREIGN KEY), id товара (FOREIGN KEY), дата заказа и статус
+заказа.
+• Таблица товаров должна содержать следующие поля: id (PRIMARY KEY), название,
+описание и цена.
+
+Создайте модели pydantic для получения новых данных и возврата существующих в
+БД для каждой из трёх таблиц (итого шесть моделей).
+Реализуйте CRUD операции для каждой из таблиц через создание маршрутов,
+REST API (итого 15 маршрутов).
+* Чтение всех
+* Чтение одного
+* Запись
+* Изменение
+* Удаление
+
+"""
+
+import databases
+import sqlalchemy
+
+from fastapi import FastAPI, Request
+from fastapi.responses import HTMLResponse
+from pydantic import BaseModel, Field
+import pandas as pd
+from fastapi.templating import Jinja2Templates
+from typing import List
+
+DATABASE_URL = "sqlite:///onlineshop.db"
+
+database = databases.Database(DATABASE_URL)
+metadata = sqlalchemy.MetaData()
+
+users = sqlalchemy.Table(
+    "users",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("name", sqlalchemy.String(40)),
+    sqlalchemy.Column("surname", sqlalchemy.String(40)),
+    sqlalchemy.Column("email", sqlalchemy.String(120)),
+    sqlalchemy.Column("password", sqlalchemy.String(5)),
+)
+
+goods = sqlalchemy.Table(
+    "goods",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("name", sqlalchemy.String(40)),
+    sqlalchemy.Column("description", sqlalchemy.String(200)),
+    sqlalchemy.Column("price", sqlalchemy.Float),
+
+)
+
+orders = sqlalchemy.Table(
+    "orders",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("user_id", sqlalchemy.ForeignKey(users.c.id)),
+    sqlalchemy.Column("goods_id", sqlalchemy.ForeignKey(goods.c.id)),
+    sqlalchemy.Column("date", sqlalchemy.String(40)),
+    sqlalchemy.Column("status", sqlalchemy.String(40)),
+
+)
+
+engine = sqlalchemy.create_engine(
+    DATABASE_URL, connect_args={"check_same_thread": False})
+metadata.create_all(engine)
+
+app = FastAPI()
+templates = Jinja2Templates(directory="templates")
+
+
+class User(BaseModel):
+    id: int
+    name: str = Field(max_length=40)
+    surname: str = Field(max_length=40)
+    email: str = Field(max_length=120)
+    password: str = Field(max_length=5)
+
+
+class UserIn(BaseModel):
+    name: str = Field(max_length=40)
+    surname: str = Field(max_length=40)
+    email: str = Field(max_length=120)
+    password: str = Field(max_length=5)
```

Comment:
Заметьте, что модели практически идентичны, просто у одной и есть дополнительное поле. Почему тогда не наследоваться?
```suggestion
class UserIn(BaseModel):
    name: str = Field(max_length=40)
    surname: str = Field(max_length=40)
    email: str = Field(max_length=120)
    password: str = Field(max_length=5)
    
    
class User(UserIn):
    id: int
```